### PR TITLE
feat: add more rules

### PR DESCRIPTION
**Pull request by DevYuns**
### Description

Add more rules. Please check the comment for each rule.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -43,7 +43,14 @@ module.exports = {
...
         '@typescript-eslint/explicit-module-boundary-types': 'off',
-        '@typescript-eslint/no-unused-vars': 'warn',
+        '@typescript-eslint/no-unused-vars': [
+          'warn',
+          {
+            varsIgnorePattern: '^_',
+            argsIgnorePattern: '^_',
+            ignoreRestSiblings: true,
+          },
+        ],
```
Comment (DevYuns) [inline index.js:61]: when add `_` character at the beginning of the variable name, eslint should ignore this rule.


```ts
function test(argA, argB) { // argB may show a warning due to unused
  const test = argA
  ...
}

function test(argA, _argB) { // _argB is not showing a warning
  const test = argA
  ...
}

```

File: index.js [hunk 2]
```diff
@@ -124,7 +131,9 @@ module.exports = {
     'object-curly-spacing': ['error', 'never'],
     'no-nested-ternary': 'warn',
     'multiline-comment-style': 'warn',
-    "spaced-comment": ["warn", "always", { "exceptions": ["-", "+"] }]
+    "spaced-comment": ["warn", "always", { "exceptions": ["-", "+"] }],
+    'no-console': 'warn',
```
Comment (DevYuns) [inline index.js:143]: To show warning when use `console`

File: index.js [hunk 3]
```diff
@@ -124,7 +131,9 @@ module.exports = {
...
+    'default-param-last': 'warn',
```
Comment (DevYuns) [inline index.js:144]: default param should be in last position. Please check this [link](https://eslint.org/docs/latest/rules/default-param-last)

File: index.js [hunk 4]
```diff
@@ -12,6 +12,13 @@ module.exports = {
       parser: '@typescript-eslint/parser',
       plugins: ['@typescript-eslint/eslint-plugin'],
       rules: {
+        "@typescript-eslint/consistent-type-imports": [
```
Comment (DevYuns) [inline index.js:15]: This is to use `import type` statement. typescript support inline type keyword from 4.5. Refer [here](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#type-on-import-names)


Rule ref : https://typescript-eslint.io/rules/consistent-type-imports/

File: index.js [hunk 5]
```diff
@@ -134,6 +134,8 @@ module.exports = {
     "spaced-comment": ["warn", "always", { "exceptions": ["-", "+"] }],
     'no-console': 'warn',
     'default-param-last': 'warn',
+    "curly": ['warn', "all"],
+    'prefer-arrow-callback': 'warn',
```
Comment (DevYuns) [inline index.js:146]: This is prefer like this. 
```ts
foo(function(a) { return a; }); // ERROR
// prefer: foo(a => a)

foo(function() { return this.a; }.bind(this)); // ERROR
// prefer: foo(() => this.a)
```

ref: https://eslint.org/docs/latest/rules/prefer-arrow-callback

File: index.js [hunk 6]
```diff
@@ -134,6 +134,8 @@ module.exports = {
     "spaced-comment": ["warn", "always", { "exceptions": ["-", "+"] }],
     'no-console': 'warn',
     'default-param-last': 'warn',
+    "curly": ['warn', "all"],
```
Comment (DevYuns) [inline index.js:145]: This is to add `{}` for if statement.

```ts
// warn
if (conditionA) return 

// prefer
if(conditionA) {
  return
}
```

ref: https://eslint.org/docs/latest/rules/curly#rule-details

**Review by JeffGuKang — APPROVED** [review]
LGTM. 좋습니다!!

**Pull request merged**